feat: sbt-ci-release

SemyonSinchenko opened the pull request:
### What changes were proposed in this pull request?
- cbt-ci-release
- updates in `build.sbt`
- new `databricks-connect` subproject

### Why are the changes needed?

Part of #510

james-willis commented:
Is there an issue for why we need databricks-specific logic? should we link that explanation here in the PR?

SemyonSinchenko commented:
> Is there an issue for why we need databricks-specific logic? should we link that explanation here in the PR?

https://github.com/mrpowers-io/tsumugi-spark/issues/55#issuecomment-2872732754

MrPowers commented:
LGTM!

rjurney reviewed (approved):
lgtm

SemyonSinchenko commented:
I'm waiting for the approve from @dmatrix about can we use the word "databricks" in the name of released artifacts, code and documentation.

WeichenXu123 commented:
> @WeichenXu123 Hello! Do you know, can we use the word "databricks" in the name of published artifacts (`graphframes-databricks-connect`) without violation of the trademark? Is there a better way to allow databricks users to use graphframes on "shared access" clusters?
> 
> cc: @rjurney

I suggest to avoid using "databricks" in the name, but using name like `graphframes-spark-connect`.

but I think supporting Spark connect is not sufficient to make it work on Databricks shared cluster, we also need to whitelist a bunch of JVM methods in the lib for running on Databricks shared cluster (and needs our security team's approval for doing that)

Kimahriman commented:
Does the API change from protobuf.Any to Array[Byte] in Spark 4 fix the shading/binary compatibility issue, at least in Spark 4+? Might be a good compromise if it does to prevent a dedicated artifact for a closed source fork

SemyonSinchenko commented:
@james-willis @Kimahriman @rjurney @dmatrix 

Hello! After some discussions with engineers from Databricks and based on your comments I did the following updates:

1. I removed `databricksConnect` as a separate artifact
2. I changed the proto shading pattern from literal to variable
3. Anyone who wants a binary compatibility with Databricks Runtimes can build GraphFrames Connect with a command `./build/sbt -Dvendor.name=dbx connect/assembly`

Please, take a look on the PR when you have time.
P.S. As I already mentioned, I'm not an experienced person in the topic of publishing anything to the sonatype central and this is my first experience with it. Please keep this in mind during the review. Thanks!

james-willis reviewed (approved):
will making users compile from source for dbx impede adoption?

SemyonSinchenko commented:
> will making users compile from source for dbx impede adoption?

Running Spark Connect plugins on Databricks is not an easy task anyway. I think we should be fine. Spark 4.0 should resolve this.
